### ISSUE TITLE
Cross compile error with cmake version 3.8.1

### ISSUE THREAD

**Issue by butterl** [iovisor/bcc#1184]
I tried follow  the cross build method at [https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwjdqLizgfbTAhUr_IMKHe9TBxMQFggnMAA&url=http%3A%2F%2Felinux.org%2Fimages%2F3%2F32%2FELC_2017_NA_dynamic_tracing_tools_on_arm_aarch64_platform.pdf&usg=AFQjCNHhtWA8v05_fafCPSLMInyiF0DI3A&sig2=tTNS8DjTUcylWDNhhU2BSg](ELC2017 presentation by Hiroyuki ISHII) 

but I got build error like below ,after upgrade cmake to version 3.8.1

~/bcc/build$ export sysroot=~/bcc ;
cmake .. -DCMAKE_CROSSCOMPILING=1 -DCMAKE_SYSROOT=${sysroot}\
-DCMAKE_FIND_ROOT_PATH=${sysroot}\
-DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER\
-DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY\
-DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY\
-DCMAKE_INSTALL_PREFIX=${sysroot}/usr\
-DCMAKE_INSTALL_LIBDIR=lib64\
-DCMAKE_C_COMPILER=aarch64-poky-linux-gcc\
-DCMAKE_CXX_COMPILER=aarch64-poky-linux-g++\
-DCMAKE_C_FLAGS="-mabi=lp64 -march=armv8-a -mtune=cortex-a57.cortex-a53 -Wall --sysroot=${sysroot}"\
-DCMAKE_CXX_FLAGS="-mabi=lp64 -march=armv8-a -mtune=cortex-a57.cortex-a53 -Wall -std=c++11 --sysroot=${sysroot}"\
-DCMAKE_EXE_LINKER_FLAGS="--sysroot=${sysroot}"\
-DCMAKE_SHARED_LINKER_FLAGS="--sysroot=${sysroot}"\
-DMULTILIB_LIB64=ON\
-DCMAKE_BUILD_TYPE=Release

-- Latest recognized Git tag is v0.3.0
-- Git HEAD is 2923e204266c8c0114b7d0572c051c655b3cc26a
-- Revision is 0.3.0-2923e204
-- Found LLVM: /usr/lib/llvm-3.7/include
CMake Warning at tests/python/CMakeLists.txt:6 (message):
  Recommended test program 'arping' not found

CMake Warning at tests/python/CMakeLists.txt:10 (message):
  Recommended test program 'netperf' not found

CMake Warning at tests/python/CMakeLists.txt:14 (message):
  Recommended test program 'iperf' not found

-- Configuring done
-- Generating done
-- Build files have been written to: /home/butter/bcc/build
butter@desktop:~/bcc/build$ make
[  1%] Building CXX object src/cc/frontends/clang/CMakeFiles/clang_frontend.dir/loader.cc.o
/home/butter/bcc/src/cc/frontends/clang/loader.cc:17:15: fatal error: map: No such file or directory
 #include <map>
               ^
compilation terminated.
make[2]: *** [src/cc/frontends/clang/CMakeFiles/clang_frontend.dir/loader.cc.o] Error 1
make[1]: *** [src/cc/frontends/clang/CMakeFiles/clang_frontend.dir/all] Error 2
make: *** [all] Error 2

Is there a max cmake version limit to bcc?

**Comment by tuilagio** [comment]
Hi,
I am working with the Beaglebone Black and having problem compiling BCC on the board too (not enough memory killed the compiling process) so I am looking for cross compiling solution. Did you make it work on your machine?

**Comment by yonghong-song** [comment]
@tuilagio No. bcc currently does not support cross compilation. You are welcome to take a look at this.

**Comment by tuilagio** [comment]
@yonghong-song
Are you sure about cross compiling not supported? What did you do with your project to move forward?

**Comment by yonghong-song** [comment]
I am not aware of anybody actually doing cross compiling with bcc. Since you have a need, maybe you take a look?
